modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/model/listener/test/CompanyModelListenerTest.java

```diff
@@ -0,0 +1,252 @@
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.security.auth.CompanyThreadLocal;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.CompanyTestUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.UnicodePropertiesBuilder;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.vulcan.util.LocalizedMapUtil;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Paulo Albuquerque
+ */
+@RunWith(Arquillian.class)
+public class CompanyModelListenerTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws Exception {
+		Company company = CompanyTestUtil.addCompany();
+
+		_companyId = company.getCompanyId();
+
+		_user = UserTestUtil.getAdminUser(_companyId);
+	}
+
+	@Test
+	public void testOnBeforeDelete() throws Exception {
+		long originalCompanyId = CompanyThreadLocal.getCompanyId();
+
+		CompanyThreadLocal.setCompanyId(_companyId);
+
+		NotificationContext notificationContext = _createNotificationContext();
+
+		NotificationTemplate notificationTemplate =
+			_notificationTemplateLocalService.addNotificationTemplate(
+				notificationContext);
+
+		Assert.assertEquals(_companyId, notificationTemplate.getCompanyId());
+
+		NotificationRecipient notificationRecipient =
+			notificationTemplate.getNotificationRecipient();
+
+		Assert.assertEquals(_companyId, notificationRecipient.getCompanyId());
+
+		List<NotificationRecipientSetting> notificationRecipientSettings =
+			notificationRecipient.getNotificationRecipientSettings();
+
+		for (NotificationRecipientSetting notificationRecipientSetting :
+				notificationRecipientSettings) {
+
+			Assert.assertEquals(
+				_companyId, notificationRecipientSetting.getCompanyId());
+		}
+
+		ObjectDefinition objectDefinition =
+			ObjectDefinitionTestUtil.publishObjectDefinition(
+				"A" + RandomTestUtil.randomString(),
+				Collections.singletonList(
+					new TextObjectFieldBuilder(
+					).labelMap(
+						LocalizedMapUtil.getLocalizedMap(
+							RandomTestUtil.randomString())
+					).name(
+						"a" + RandomTestUtil.randomString()
+					).build()),
+				ObjectDefinitionConstants.SCOPE_COMPANY, _user.getUserId());
```

Comment:
There is a method in ObjectDefinitionTestUtil.publishObjectDefinition() that you don't need to pass the first parameter because it already creates a random name for you:
```
public static ObjectDefinition publishObjectDefinition(
			List<ObjectField> objectFields, String scope, long userId)
```